test_mpop_service/TestMpopService.cpp

```diff
@@ -443,3 +438,116 @@ void TestMpopService::test_20_setUserAnswer() {
     QCOMPARE(answers1[TEST_QUESTION_03_ID], TEST_QUESTION_03_UPDATE_VALUE);
     QCOMPARE(answers1[TEST_QUESTION_04_ID], TEST_QUESTION_04_UPDATE_VALUE);
 }
+
+void TestMpopService::test_22_getAnswerByAge() {
+
+    if (! this->is_mysql_supported) {
+        QSKIP("This test requires MySQL");
+    }
+
+    // User Question
+    static const QString TEST_QUESTION_05_ID = "equitable_vulnerables";
+
+    // User Answers
+    static const int TEST_QUESTION_05_VALUE = 35;
+
+    // Calculated Avg. for three User
+    static const int Test_QUESTION_05_AVG = 35;
+
+    // declaration of RFIDs
+    static const QString TEST_RFID_TAG_1 = "test_RFID_tag_0009";
+
+    // Get or create user id.
+    int user_1 = this->facade->getOrCreateUser(TEST_RFID_TAG_1);
+
+    // set Users Answers
+    this->facade->setUserAnswer(user_1, TEST_QUESTION_05_ID, TEST_QUESTION_05_VALUE);
+
+
+    QList<int> answers1 = this->facade->getAnswerByAge(TEST_QUESTION_05_ID,"all","all","all");
+
+    qDebug() << "Questions List" << answers1;
+    //QCOMPARE(answers1[TEST_QUESTION_05_ID], Test_QUESTION_05_AVG);
+
+}
+
+void TestMpopService::test_23_getAnswerByEthnicity() {
+    qDebug() << "test_23_getAnswerByEthnicity";
+    // Set demographic questions for multiple user
+    // Set multiple answers for multiple user
+    // check avg value of each question is ok.
+
+    if (! this->is_mysql_supported) {
+        QSKIP("This test requires MySQL");
+    }
+
+    // User Questions
+    static const QString TEST_QUESTION_06_ID = "equitable_jeunes_contrevenants";
+
+    // User Answers
+    static const int TEST_QUESTION_06_VALUE = 45;
+
+    // Calculated Avg. for three User
+    static const int Test_QUESTION_06_AVG = 45;
+
+    // declaration of RFIDs
+    static const QString TEST_RFID_TAG_1 = "test_RFID_tag_00010";
+
+    // declaration of Ethnicity for users
+    static const QString TEST_USER_1_Ethnicity = "quebecer";
+
+    // Get or create user id.
+    int user_1 = this->facade->getOrCreateUser(TEST_RFID_TAG_1);
+
+
+    // set users Ethnicity
+    bool okSetEthnicity_1 = this->facade->setUserEthnicity(user_1, TEST_USER_1_Ethnicity);
+    QCOMPARE(okSetEthnicity_1, true);
+
+    // set Users Answers
+    this->facade->setUserAnswer(user_1, TEST_QUESTION_06_ID, TEST_QUESTION_06_VALUE);
```

Comment:
That's good. It would be even better if we'd create a dozen of visitors, and all set their answer for the same question, and then check what we check below.